confidential-data-hub/storage/src/volume_type/blockdevice/mod.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2024 Intel
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+pub mod error;
+pub mod luks;
+
+use super::SecureMount;
+use async_trait::async_trait;
+use error::{BlockDeviceError, Result};
+use kms::{Annotations, ProviderSettings};
+use log::{debug, error};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use strum::{Display, EnumString};
+
+#[derive(EnumString, Serialize, Deserialize, Display, Debug, PartialEq, Eq)]
+pub enum BlockDeviceEncryptType {
+    #[strum(serialize = "luks")]
+    LUKS,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub struct BlockDeviceParameters {
+    /// The device number, formatted as "MAJ:MIN".
+    #[serde(rename = "deviceId")]
+    pub device_id: String,
+
+    /// The encryption type. Currently, only LUKS is supported.
+    #[serde(rename = "encryptType")]
+    pub encryption_type: BlockDeviceEncryptType,
+
+    /// Encryption key. If not set, generate a random 4096-byte key
+    #[serde(rename = "encryptKey")]
+    pub encryption_key: Option<String>,
+
+    /// Indicates whether to enable dm-integrity.
+    #[serde(rename = "dataIntegrity")]
+    pub data_integrity: String,
+}
+pub(crate) struct BlockDevice;
+
+#[async_trait]
+pub trait Encryptor {
```

Comment:
should this be encryptor or decryptor?

Reply:
For trusted storage, this should be encryptor. The guest will need to follow three steps: encrypt the block device from the host as a LUKS-encrypted block device, open the encrypted device with the key, and mount it. However, for secure storage, the guest will only need to perform the latter two steps. Overall, I think `encryptor` or `decryptor` are not suitable names for it. Therefore, I have renamed it to `Interpreter`.